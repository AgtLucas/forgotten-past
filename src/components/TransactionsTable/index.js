import React from 'react';
import Dinero from 'dinero.js';
import Button from '../Button';
import styles from './styles.module.css';

const TRANSACTION_TYPE = {
  'credit': 'Crédito',
  'debit': 'Débito',
};

const BRL = value =>
  Dinero({ amount: value, currency: 'BRL' })
    .setLocale('pt-BR')
    .toFormat('$0,0.00');


const TransactionsTable = ({ transactions, transactionsTotalPrice, deleteTransactionById }) => {
  if (!transactions || !transactions.length) return (
    <p className={styles.disclaimer}>
      Você ainda não cadastrou nenhuma transação
    </p>
  );

  return (
    <table className={styles.table}>
      <thead>
      <tr>
        <th>Descrição</th>
        <th>Preço</th>
        <th>Tipo de transação</th>
        <th></th>
      </tr>
      </thead>
      <tfoot>
      <tr>
        <th colSpan={4}>Total: {BRL(transactionsTotalPrice)}</th>
      </tr>
      </tfoot>
      <tbody>
      {transactions.map((transaction, i) => (
        <tr key={i}>
          <td>{transaction.description}</td>
          <td>{BRL(parseInt(transaction.price))}</td>
          <td>{TRANSACTION_TYPE[transaction.transactionType]}</td>
          <td style={{ width: '30px' }}>
            <Button
              variant='emoji'
              onClick={() => deleteTransactionById(transaction.id)}
              title={`Deletar transação: ${transaction.description}`}
            >
              <span role='img' aria-label={`Deletar transação: ${transaction.description}`}>
                ❌
              </span>
            </Button>
          </td>
        </tr>
      )).reverse()}
      </tbody>
    </table>
  )
};

export default TransactionsTable;