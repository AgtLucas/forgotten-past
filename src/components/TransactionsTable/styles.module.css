.table {
  margin: 0 20px;
}

.disclaimer {
  text-align: center;
}