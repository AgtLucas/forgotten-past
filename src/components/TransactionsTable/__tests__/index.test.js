import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TransactionsTable from '../';

describe('TransactionsTable', () => {
  it('should render TransactionsTable', () => {
    const props = {
      transactions: [
        {
          description: 'Foo',
          price: 100,
          type: 'credit'
        },
      ],
      transactionsTotalPrice: 231,
      deleteTransactionById: jest.fn(),
    };

    const wrapper = shallow(<TransactionsTable {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});