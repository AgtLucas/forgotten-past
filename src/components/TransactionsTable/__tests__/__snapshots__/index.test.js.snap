// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsTable should render TransactionsTable 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Descrição
      </th>
      <th>
        Preço
      </th>
      <th>
        Tipo de transação
      </th>
      <th />
    </tr>
  </thead>
  <tfoot>
    <tr>
      <th
        colSpan={4}
      >
        Total: 
        R$2.31
      </th>
    </tr>
  </tfoot>
  <tbody>
    <tr
      key="0"
    >
      <td>
        Foo
      </td>
      <td>
        R$1.00
      </td>
      <td />
      <td
        style={
          Object {
            "width": "30px",
          }
        }
      >
        <Button
          onClick={[Function]}
          title="Deletar transação: Foo"
          variant="emoji"
        >
          <span
            aria-label="Deletar transação: Foo"
            role="img"
          >
            ❌
          </span>
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;
